### ISSUE TITLE
Lint the `__all__` variable for validity

### ISSUE THREAD

**Issue by hwittenborn** [astral-sh/ruff#17794]
### Summary

Ruff doesn't complain about this non-existent `wow` import, even though it doesn't exist:

```python
__all__ = ["ai", "env", "tools", "endpoints", "wow"]
from . import ai, env, tools, endpoints
```

Would it be possible to lint `__all__` for non-existent members, in the same way it checks for members that _aren't_ present?

**Comment by Daverball** [comment]
It already does, it emits an [`undefined-name`](https://docs.astral.sh/ruff/rules/undefined-name/) (`F822`) for your example. If it doesn't you either have that diagnostic disabled or your stripped down example isn't enough on its own to reproduce your problem.

**Comment by Daverball** [comment]
`F` should be part of the default configuration, so it should be enabled. Does `ruff check --isolated` give you the same output? If not there may be a config file somewhere that's getting picked up by ruff. You can use `--show-settings` to see what the individual settings currently resolve to when checking any given file.

See [Config File Discovery](https://docs.astral.sh/ruff/configuration/#config-file-discovery) for where those settings could come from.
